# Business Understanding

Se pretende explicar qué factores afectan significativamente el precio de las casas en el condado King de Estados Unidos

# Data Understanding

In [8]:
# Cargar paquetes
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [9]:
df = pd.read_csv('../kc_house_data.csv')

In [10]:
df.info ()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
id               21613 non-null int64
date             21613 non-null object
price            21613 non-null float64
bedrooms         21613 non-null int64
bathrooms        21613 non-null float64
sqft_living      21613 non-null int64
sqft_lot         21613 non-null int64
floors           21613 non-null float64
waterfront       21613 non-null int64
view             21613 non-null int64
condition        21613 non-null int64
grade            21613 non-null int64
sqft_above       21613 non-null int64
sqft_basement    21613 non-null int64
yr_built         21613 non-null int64
yr_renovated     21613 non-null int64
zipcode          21613 non-null int64
lat              21613 non-null float64
long             21613 non-null float64
sqft_living15    21613 non-null int64
sqft_lot15       21613 non-null int64
dtypes: float64(5), int64(15), object(1)
memory usage: 3.5+ MB


In [11]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view     ...      grade  sqft_above  \
0      5650     1.0           0     0     ...          7        1180   
1      7242     2.0           0     0     ...          7        2170   
2     10000     1.0           0     0     ...          6         770   
3      5000     1.0           0     0     ...          7        1050   
4      8080     1.0           0     0     ...          8        1680   

   sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
0              0      1955             0    98178  47.5112 -122.257   
1            400      1951          1991    98125  47.7210 -122.319   
2              0      1933             0    98028  47.7379 -122.233   
3            910      1965             0    98136  47.5208 -122.393   
4              0      1987             0    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503  

[5 rows x 21 columns]

In [18]:
df.describe().round(2)

id       price  bedrooms  bathrooms  sqft_living    sqft_lot  \
count  2.161300e+04    21613.00  21613.00   21613.00     21613.00    21613.00   
mean   4.580302e+09   540088.14      3.37       2.11      2079.90    15106.97   
std    2.876566e+09   367127.20      0.93       0.77       918.44    41420.51   
min    1.000102e+06    75000.00      0.00       0.00       290.00      520.00   
25%    2.123049e+09   321950.00      3.00       1.75      1427.00     5040.00   
50%    3.904930e+09   450000.00      3.00       2.25      1910.00     7618.00   
75%    7.308900e+09   645000.00      4.00       2.50      2550.00    10688.00   
max    9.900000e+09  7700000.00     33.00       8.00     13540.00  1651359.00   

         floors  waterfront      view  condition     grade  sqft_above  \
count  21613.00    21613.00  21613.00   21613.00  21613.00    21613.00   
mean       1.49        0.01      0.23       3.41      7.66     1788.39   
std        0.54        0.09      0.77       0.65      1.18      828.09   
min        1.00        0.00      0.00       1.00      1.00      290.00   
25%        1.00        0.00      0.00       3.00      7.00     1190.00   
50%        1.50        0.00      0.00       3.00      7.00     1560.00   
75%        2.00        0.00      0.00       4.00      8.00     2210.00   
max        3.50        1.00      4.00       5.00     13.00     9410.00   

       sqft_basement  yr_built  yr_renovated   zipcode       lat      long  \
count       21613.00  21613.00      21613.00  21613.00  21613.00  21613.00   
mean          291.51   1971.01         84.40  98077.94     47.56   -122.21   
std           442.58     29.37        401.68     53.51      0.14      0.14   
min             0.00   1900.00          0.00  98001.00     47.16   -122.52   
25%             0.00   1951.00          0.00  98033.00     47.47   -122.33   
50%             0.00   1975.00          0.00  98065.00     47.57   -122.23   
75%           560.00   1997.00          0.00  98118.00     47.68   -122.12   
max          4820.00   2015.00       2015.00  98199.00     47.78   -121.32   

       sqft_living15  sqft_lot15  
count       21613.00    21613.00  
mean         1986.55    12768.46  
std           685.39    27304.18  
min           399.00      651.00  
25%          1490.00     5100.00  
50%          1840.00     7620.00  
75%          2360.00    10083.00  
max          6210.00   871200.00

# Data Preparation

In [19]:
# Elimino las variables ID y DATE para quedarme solo con las variables que son numéricas.
del df['id']
del df['date']

In [21]:
df.head()

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  waterfront  \
0  221900.0         3       1.00         1180      5650     1.0           0   
1  538000.0         3       2.25         2570      7242     2.0           0   
2  180000.0         2       1.00          770     10000     1.0           0   
3  604000.0         4       3.00         1960      5000     1.0           0   
4  510000.0         3       2.00         1680      8080     1.0           0   

   view  condition  grade  sqft_above  sqft_basement  yr_built  yr_renovated  \
0     0          3      7        1180              0      1955             0   
1     0          3      7        2170            400      1951          1991   
2     0          3      6         770              0      1933             0   
3     0          5      7        1050            910      1965             0   
4     0          3      8        1680              0      1987             0   

   zipcode      lat     long  sqft_living15  sqft_lot15  
0    98178  47.5112 -122.257           1340        5650  
1    98125  47.7210 -122.319           1690        7639  
2    98028  47.7379 -122.233           2720        8062  
3    98136  47.5208 -122.393           1360        5000  
4    98074  47.6168 -122.045           1800        7503

In [24]:
#Correlacion entre las variables que utilizaré en mi modelo

df[['price', 'sqft_living', 'sqft_lot', 'waterfront', 'view']].corr()

price  sqft_living  sqft_lot  waterfront      view
price        1.000000     0.702035  0.089661    0.266369  0.397293
sqft_living  0.702035     1.000000  0.172826    0.103818  0.284611
sqft_lot     0.089661     0.172826  1.000000    0.021604  0.074710
waterfront   0.266369     0.103818  0.021604    1.000000  0.401857
view         0.397293     0.284611  0.074710    0.401857  1.000000

In [25]:
# Genero una nueva variable con la interseccion pie cuadrado habitable con frente al agua
df['living_water'] = df['sqft_living'] * df['waterfront']
# Genero una nueva variable con la intersecion pie cuadrado del lote con frente al agua
df['lot_water'] = df['sqft_lot'] * df['waterfront']

In [28]:
df[['sqft_living', 'sqft_lot','waterfront','living_water', 'lot_water' ]] .head()

sqft_living  sqft_lot  waterfront  living_water  lot_water
0         1180      5650           0             0          0
1         2570      7242           0             0          0
2          770     10000           0             0          0
3         1960      5000           0             0          0
4         1680      8080           0             0          0

# Modeling

Yo quiero saber cuando aumenta más un pie cuadrado de construcción frente al mar con respecto a no tener esta vista.

In [29]:
# Selecciono mis variables
y = df.pop('price')
x = df[['sqft_living', 'sqft_lot','waterfront','living_water', 'lot_water' ]]

In [33]:
# Cargar modelo
model = sm.OLS(y, x).fit()

# Resumen del modelo
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.857
Model:                            OLS   Adj. R-squared:                  0.857
Method:                 Least Squares   F-statistic:                 2.596e+04
Date:                Fri, 11 Jan 2019   Prob (F-statistic):               0.00
Time:                        15:39:13   Log-Likelihood:            -2.9901e+05
No. Observations:               21613   AIC:                         5.980e+05
Df Residuals:                   21608   BIC:                         5.981e+05
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
sqft_living    259.7233      0.805    322.730      0.000     258.146     261.301
sqft_lot        -0.2536      0.041     -6.139      0.000      -0.335      -0.173
waterfront   -2.463e+05   4.42e+04     -5.575      0.000   -3.33e+05    -1.6e+05
living_water   354.3594     12.100     29.287      0.000     330.643     378.076
lot_water       -1.3507      0.453     -2.979      0.003      -2.239      -0.462
==============================================================================
Omnibus:                    13731.011   Durbin-Watson:                   1.979
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           477998.672
Skew:                           2.529   Prob(JB):                         0.00
Kurtosis:                      25.477   Cond. No.                     1.16e+06
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.16e+06. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

# Evaluation

Se encontró lo siguiente:
1. Caeteris Paribus; al aumentar un pie cuadrado habitable en la casa a comprar, se incrementa el precio de la vivienda en 259.72 dólares.
2. Caeteris Paribus; al aumentar un pie cuadrado en el terreno de la casa a comprar, el precio de la vivienda disminuye -0.2536 dólares.
3. Caeteris Paribus; al aumentar un pie cuadrado habitable de la casa a comprar y que esta esté frente al mar, se incrementa el precio de la vivienda 354.35 dólares más con respecto a las que no están frente al mar.
4. Caeteris Paribus; al aumentar un pie cuadrado de terreno de la casa a comprar y que esté frente al mar, el precio de la vivienda disminuye 1.35 dólares con respecto a las que no están frente al mar.

Asimismo, es importante resaltar que se tiene una R cuadrada ajustada de .857 y todos los coeficientes son estadisticamente significativos.